### PR TITLE
Workaround XUnit FileLoadException issues

### PR DESCRIPTION
**Pull request by echesakovMSFT**
To workaround XUnit FileLoadException issues use custom xunit.console.dll (2.4.1-pre-4059) with the following changes https://github.com/xunit/xunit/pull/1846 https://github.com/echesakovMSFT/xunit/tree/UseConcurrentDictionaryInDependencyContextAssemblyCache

**Related issues:** https://github.com/dotnet/coreclr/issues/20392 https://github.com/dotnet/coreclr/issues/20594 https://github.com/xunit/xunit/issues/1842

@RussKeldorph @BruceForstall @jashook PTAL

### CODE REVIEW

**Comment by echesakovMSFT** [comment]
@dotnet-bot test this

**Comment by echesakovMSFT** [comment]
@dotnet-bot test this

**Comment by echesakovMSFT** [comment]
`java.nio.channels.ClosedChannelException`

@dotnet-bot test Windows_NT arm Cross Debug Innerloop Build
@dotnet-bot test Windows_NT arm64 Cross Checked Innerloop Build and Test

**Comment by echesakovMSFT** [comment]
java.nio.channels.ClosedChannelException again

@dotnet-bot test OSX10.12 x64 Checked Innerloop Build and Test

**Comment by echesakovMSFT** [comment]
java.nio.channels.ClosedChannelException again
@dotnet-bot test OSX10.12 x64 Checked Innerloop Build and Test

**Comment by echesakovMSFT** [comment]
java.nio.channels.ClosedChannelException again in tst job

@dotnet-bot test Ubuntu16.04 arm64 Cross Checked no_tiered_compilation_innerloop Build and Test

**Comment by echesakovMSFT** [comment]
This one is got stuck

test Ubuntu16.04 arm64 Cross Checked Innerloop Build and Test

**Comment by echesakovMSFT** [comment]
@BruceForstall I added the comment, can you please take a look?

File: tests/runtest.py [hunk 1]
```diff
@@ -764,6 +765,58 @@ def run_tests(host_os,
...
+    #  * https://github.com/dotnet/coreclr/issues/20594
+    #  * https://github.com/xunit/xunit/issues/1842 
+    #  * https://github.com/xunit/xunit/pull/1846
+    #
+    #=====================================================================================================================================================
+
+    print("Download and overwrite xunit.console.dll in Core_Root")
+
+    try:
+        import urllib.request
```
Comment (jashook) [inline tests/runtest.py]: I would much rather prefer this be wrapped in a if python version < 3 then catching the exception. We will eventually need to remove these python2 isms and this setup makes that work more difficult.

Comment (jashook) [reply tests/runtest.py]: I would suggest this import be conditional at the header of the file. After the last from import add something like:

```
# Version specific imports

if sys.version_info < (3,0):
    import urllib.request
else:
    import urllib
```

Comment (jashook) [reply tests/runtest.py]: Then you can add the if here as well. I am fine with changing this to a method call is_python3.

```
if sys.version_info < (3,0):
    urlretrieve = urllib.request.urlretrieve
else:
    urlretrieve = urllib.urlretrieve
```

File: tests/runtest.py [hunk 2]
```diff
@@ -764,6 +765,58 @@ def run_tests(host_os,
...
+
+    zipfilename = os.path.join(tempfile.gettempdir(), "xunit.console.dll.zip")
+    url = r"https://clrjit.blob.core.windows.net/xunit-console/xunit.console.dll.zip"
+    urlretrieve(url, zipfilename)
+
+    with zipfile.ZipFile(zipfilename,"r") as ziparch:
+        ziparch.extractall(core_root)
+
+    os.remove(zipfilename)
+
```
Comment (jashook) [inline tests/runtest.py]: Nit, extra newline

File: tests/runtest.py [hunk 3]
```diff
@@ -764,6 +765,58 @@ def run_tests(host_os,
...
+        urlretrieve = urllib.urlretrieve
+
+    zipfilename = os.path.join(tempfile.gettempdir(), "xunit.console.dll.zip")
+    url = r"https://clrjit.blob.core.windows.net/xunit-console/xunit.console.dll.zip"
+    urlretrieve(url, zipfilename)
+
+    with zipfile.ZipFile(zipfilename,"r") as ziparch:
+        ziparch.extractall(core_root)
+
+    os.remove(zipfilename)
```
Comment (jashook) [inline tests/runtest.py]: Nit: Add an assert that `!os.path.isfile(zipfilename)`

**Comment by echesakovMSFT** [comment]
java.nio.channels.ClosedChannelException

test Windows_NT arm Cross Checked Innerloop Build and Test
test Windows_NT arm64 Cross Checked Innerloop Build and Test

**Comment by echesakovMSFT** [comment]
java.nio.channels.ClosedChannelException

test Windows_NT arm Cross Checked Innerloop Build and Test

**Comment by echesakovMSFT** [comment]
test Windows_NT arm Cross Checked Innerloop Build and Test

**Pull request merged**